src/ctapipe/io/hdf5eventsource.py

```diff
@@ -70,9 +70,7 @@
 
 
 COMPATIBLE_DATA_MODEL_VERSIONS = [
-    "v4.0.0",
-    "v5.0.0",
-    "v5.1.0",
+    "v6.0.0",
```

Comment:
Since this only affects R1 and DL0 data stored in hdf5 files, I wonder if it is too harsh to make old versions completely unreadable here? Writing that level of data is more of an exception than the default case.   Of course we will likely re-process everything soon, so maybe it's not a huge issue.  However, maybe an interim solution would be to not drop support for old versions, but simply raise an exception if somebody tries to read files that have R0 or DL0 data in from an old version number, but proceed as usual for DL1 and beyond? 

I.e. right now we have a strict data version that is checked, but maybe we should check the version per data level or at least for R0/DL0 vs higher data levels?

Reply:
Yes, I think especially in this case, since the compatibility layer is extremely simple (just check ndim==2 and then do `waveform[np.newaxis, ...]` we should make the effort and keep compatibility to the older DL1+ versions at least.